File: plugins/scandit-sdk/evals/barcode-capture-net-android/fixtures/ZxingActivity.cs
using Android.Content;
using Android.OS;
using Android.Widget;
using AndroidX.AppCompat.App;
using ZXing.Mobile;

namespace MyApp;

public record ScannedBarcode(string Value, string Format);

[Activity(Label = "Scanner")]
public class ScannerActivity : AppCompatActivity
{
    private TextView resultLabel = null!;
    private MobileBarcodeScanner scanner = null!;
    private readonly List<ScannedBarcode> scannedBarcodes = new();

    protected override void OnCreate(Bundle? savedInstanceState)
    {
        base.OnCreate(savedInstanceState);
        this.SetContentView(Resource.Layout.activity_scanner);
        this.resultLabel = this.FindViewById<TextView>(Resource.Id.resultLabel)!;
        this.scanner = new MobileBarcodeScanner();
        _ = this.StartScanAsync();
    }

    private async Task StartScanAsync()
    {
        var options = new MobileBarcodeScanningOptions
        {
            PossibleFormats = new List<ZXing.BarcodeFormat>
            {
                ZXing.BarcodeFormat.EAN_13,
                ZXing.BarcodeFormat.CODE_128,
                ZXing.BarcodeFormat.QR_CODE
            },
            UseFrontCameraIfAvailable = false,
        };

        var result = await this.scanner.Scan(options);
        if (result != null && !string.IsNullOrEmpty(result.Text))
        {
            var barcode = new ScannedBarcode(result.Text, result.BarcodeFormat.ToString());
            if (!this.scannedBarcodes.Any(b => b.Value == barcode.Value))
            {
                this.scannedBarcodes.Add(barcode);
                this.resultLabel.Text = $"Last scan: {barcode.Value} ({this.scannedBarcodes.Count} total)";
            }
            _ = this.StartScanAsync();
        }
    }
}

File: plugins/scandit-sdk/evals/sparkscan-android/fixtures/ZxingActivity.kt
import android.content.Intent
import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.google.zxing.integration.android.IntentIntegrator
import com.google.zxing.integration.android.IntentResult

data class ScannedBarcode(val value: String, val format: String)

class ScannerActivity : AppCompatActivity() {

    private lateinit var resultLabel: TextView
    private val scannedBarcodes = mutableListOf<ScannedBarcode>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_scanner)
        resultLabel = findViewById(R.id.resultLabel)
        startScan()
    }

    private fun startScan() {
        val integrator = IntentIntegrator(this)
        integrator.setDesiredBarcodeFormats(
            IntentIntegrator.EAN_13,
            IntentIntegrator.CODE_128,
            IntentIntegrator.QR_CODE
        )
        integrator.setPrompt("Scan a barcode")
        integrator.setCameraId(0)
        integrator.setBeepEnabled(true)
        integrator.setBarcodeImageEnabled(false)
        integrator.initiateScan()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        val result: IntentResult = IntentIntegrator.parseActivityResult(requestCode, resultCode, data)
        if (result.contents != null) {
            val barcode = ScannedBarcode(result.contents, result.formatName ?: "UNKNOWN")
            if (scannedBarcodes.none { it.value == barcode.value }) {
                scannedBarcodes.add(barcode)
                resultLabel.text = "Last scan: ${barcode.value} (${scannedBarcodes.size} total)"
            }
            startScan()
        } else {
            super.onActivityResult(requestCode, resultCode, data)
        }
    }
}
